addons/hr_timesheet/views/project_task_views.xml

```diff
@@ -212,8 +212,8 @@
             <field name="priority">10</field>
             <field name="arch" type="xml">
                 <xpath expr="//filter[@name='blocking']/following-sibling::separator[1]" position="after">
-                    <filter string="Timesheets 80%" name="timesheet_80" domain="[('remaining_hours_percentage', '&gt;', 0.0), ('remaining_hours_percentage', '&lt;=', 0.2)]"/>
-                    <filter string="Timesheets &gt;100%" name="timesheet_exceeded" domain="[('overtime', '&gt;', 0)]"/>
+                    <filter string="Timesheets 80%" name="timesheet_80" invisible="not context.get('allow_timesheets', True)" domain="[('remaining_hours_percentage', '&gt;', 0.0), ('remaining_hours_percentage', '&lt;=', 0.2)]"/>
+                    <filter string="Timesheets &gt;100%" name="timesheet_exceeded" invisible="not context.get('allow_timesheets', True)" domain="[('overtime', '&gt;', 0)]"/>
```

Comment:
I think it would be slightly better to not use a truthy default value in the `get`, so that we let it be invisible unless we specifically make it visible with the context key
```suggestion
                    <filter string="Timesheets 80%" name="timesheet_80" invisible="not context.get('allow_timesheets')" domain="[('remaining_hours_percentage', '&gt;', 0.0), ('remaining_hours_percentage', '&lt;=', 0.2)]"/>
                    <filter string="Timesheets &gt;100%" name="timesheet_exceeded" invisible="not context.get('allow_timesheets')" domain="[('overtime', '&gt;', 0)]"/>
```